drivers/sensor/tdk/icm42670/icm42670.c

```diff
@@ -442,133 +793,136 @@ static int icm42670_sample_fetch_temp(const struct device *dev)
 	return 0;
 }
 
-static int icm42670_sample_fetch(const struct device *dev, enum sensor_channel chan)
+static int icm42670_fetch_from_registers(const struct device *dev, enum sensor_channel chan)
 {
-	uint8_t status;
-	const struct icm42670_config *cfg = dev->config;
+	struct icm42670_data *data = dev->data;
+	int res = 0;
+	uint8_t int_status;
 
-	icm42670_lock(dev);
+	LOG_ERR("Fetch from reg");
 
-	int res = cfg->bus_io->read(&cfg->bus, REG_INT_STATUS_DRDY, &status, 1);
+	icm42670_lock(dev);
 
-	if (res) {
-		goto cleanup;
+	/* Ensure data ready status bit is set */
+	res |= inv_imu_read_reg(&data->driver, INT_STATUS_DRDY, 1, &int_status);
+
+	if (int_status & INT_STATUS_DRDY_DATA_RDY_INT_MASK) {
+		switch (chan) {
+		case SENSOR_CHAN_ALL:
+			res |= icm42670_sample_fetch_accel(dev);
+			res |= icm42670_sample_fetch_gyro(dev);
+			res |= icm42670_sample_fetch_temp(dev);
```

Comment:
This is another example where you will end up returning something that isn't an errno anymore because you've logically OR-ed together multiple values

Reply:
Change done